### PR TITLE
signed url api added for all cloud services

### PR DESCRIPTION
**Pull request by aman-tunerlabs**

### CODE REVIEW

File: generics/services/aws.js [hunk 1]
```diff
@@ -86,8 +86,70 @@ let getDownloadableUrl = function( filePath,bucketName ) {
...
+let signedUrl = (folderPath = "", fileName = "",bucketName) =>{
+    return new Promise(async (resolve, reject) => {
+        try {
+            
+            if(folderPath == "" || fileName == "") {
+                throw new Error(httpStatusCode.bad_request.status);
+            }
+
+            let noOfMinutes = 30;
+            let expiry = 60 * noOfMinutes;
```
Comment (aks30) [inline generics/services/aws.js]: @aman-tunerlabs - Can this expiry no of minutes be a common constant across services

File: controllers/v1/cloud-services/aws.js [hunk 2]
```diff
@@ -103,5 +103,84 @@ module.exports = class Aws {
 
     }
 
+
+    /**
+     * @api {post} /kendra/api/v1/cloud-services/aws/getSignedUrls  
+     * Get signed URL.
```
Comment (aks30) [inline controllers/v1/cloud-services/aws.js:109]: @aman-tunerlabs - Change API name to, 

> Get pre-signed URLs for file upload

File: generics/services/google-cloud.js [hunk 3]
```diff
@@ -70,7 +70,60 @@ let getDownloadableUrl = ( filename,bucketName) =>{
...
+  return new Promise(async (resolve, reject) => {
+    try {
+
+      var myBucket = storage.bucket(bucketName);
+      
+      if(folderPath == "" || fileName == "") {
+        throw new Error(httpStatusCode.bad_request.status);
+      }
+      
+      let noOfMinutes = 30;
```
Comment (aks30) [inline generics/services/google-cloud.js]: @aman-tunerlabs - Can this expiry no of minutes be a common constant across services

File: controllers/v1/cloud-services/azure.js [hunk 4]
```diff
@@ -172,5 +167,84 @@ module.exports = class Azure {
 
     }
 
+     /**
+     * @api {post} /kendra/api/v1/cloud-services/azure/getSignedUrl  
+     * Get signed URL.
```
Comment (aks30) [inline controllers/v1/cloud-services/azure.js:172]: @aman-tunerlabs - Change API name to,

Get pre-signed URLs for file upload

File: controllers/v1/cloud-services/gcp.js [hunk 5]
```diff
@@ -103,5 +104,84 @@ module.exports = class Gcp {
 
     }
 
+
+    /**
+     * @api {post} /kendra/api/v1/cloud-services/gcp/getSignedUrl  
+     * Get signed URL.
```
Comment (aks30) [inline controllers/v1/cloud-services/gcp.js:110]: @aman-tunerlabs - Change API name to,

Get pre-signed URLs for file upload

File: generics/services/azure.js [hunk 6]
```diff
@@ -102,9 +102,67 @@ let getDownloadableUrl = function (filePath, containerName) {
...
+  return new Promise(async (resolve, reject) => {
+    try {
+      
+      if(folderPath == "" || fileName == "") {
+        throw new Error(httpStatusCode.bad_request.status);
+      }
+      
+      let startDate = new Date();
+      startDate.setMinutes(startDate.getMinutes() - 5);
+      let expiryDate = new Date(startDate);
```
Comment (aks30) [inline generics/services/azure.js:125]: @aman-tunerlabs - Set expiry from a common constant across services

Comment (aman-tunerlabs) [reply generics/services/azure.js:125]: @aks30  Expiry logic is different for different services. So thats why i did not kept it common

**Pull request merged**